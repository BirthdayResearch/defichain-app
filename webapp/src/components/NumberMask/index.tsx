import React from 'react';
import NumberFormat from 'react-number-format';

export interface NumberMaskProps {
  value: string;
  defaultValue?: string | number;
}

const NumberMask: React.FunctionComponent<NumberMaskProps> = (
  props: NumberMaskProps
) => {
  return (
    <NumberFormat
      value={props.value.toString()}
      defaultValue={props.defaultValue ?? 0}
      displayType={'text'}
      thousandSeparator={true}
      isNumericString={true}
    />
  );
};

export default NumberMask;
