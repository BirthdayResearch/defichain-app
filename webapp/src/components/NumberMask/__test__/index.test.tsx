import React from 'react';
import NumberMask from '../index';
import { shallow } from 'enzyme';

describe('NumberMask component', () => {
  it('should check for snapshot', () => {
    const props = {
      value: 'value',
      defaultValue: 'defaultValue',
    };
    const wrapper = shallow(<NumberMask {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
