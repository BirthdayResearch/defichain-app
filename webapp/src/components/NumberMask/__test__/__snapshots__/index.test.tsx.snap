// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberMask component should check for snapshot 1`] = `
<NumberFormat
  allowEmptyFormatting={false}
  allowLeadingZeros={false}
  allowNegative={true}
  decimalSeparator="."
  defaultValue="defaultValue"
  displayType="text"
  fixedDecimalScale={false}
  isAllowed={[Function]}
  isNumericString={true}
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseUp={[Function]}
  onValueChange={[Function]}
  prefix=""
  suffix=""
  thousandSeparator={true}
  thousandsGroupStyle="thousand"
  type="text"
  value="value"
/>
`;
